rust/phoenix-channel/src/lib.rs

```diff
@@ -698,4 +757,90 @@ mod tests {
             Payload::Message(InitMessage::Init(EmptyInit {}))
         );
     }
+
+    #[test]
+    fn unmatched_topic_reply() {
+        let actual_reply = r#"
+            {
+               "event":"phx_reply",
+               "ref":"12",
+               "topic":"client",
+               "payload":{
+                  "status":"error",
+                  "response":{
+                     "reason":"unmatched topic"
+                  }
+               }
+            }
+        "#;
+        let actual_reply: Payload<(), ()> = serde_json::from_str(actual_reply).unwrap();
+        let expected_reply = Payload::<(), ()>::Reply(ReplyMessage::PhxReply(PhxReply::Error(
+            ErrorInfo::Reason(Reason::Known(KnownError::UnmatchedTopic)),
+        )));
+        assert_eq!(actual_reply, expected_reply);
+    }
+
+    #[test]
+    fn phx_close() {
+        let actual_reply = r#"
+        {
+          "event": "phx_close",
+          "ref": null,
+          "topic": "client",
+          "payload": {}
+        }
+        "#;
+        let actual_reply: Payload<(), ()> = serde_json::from_str(actual_reply).unwrap();
+        let expected_reply = Payload::<(), ()>::ControlMessage(ControlMessage::PhxClose(Empty {}));
+        assert_eq!(actual_reply, expected_reply);
+    }
+
+    #[test]
+    fn token_expired() {
+        let actual_reply = r#"
+        {
+          "event": "disconnect",
+          "ref": null,
+          "topic": "client",
+          "payload": { "reason": "token_expired" }
+        }
+        "#;
+        let actual_reply: Payload<(), ()> = serde_json::from_str(actual_reply).unwrap();
+        let expected_reply = Payload::<(), ()>::ControlMessage(ControlMessage::Disconnect {
+            reason: "token_expired".to_string(),
+        });
+        assert_eq!(actual_reply, expected_reply);
+    }
+
+    #[test]
+    fn unexpected_error_reply() {
+        let actual_reply = r#"
+            {
+               "event":"phx_reply",
+               "ref":"12",
+               "topic":"client",
+               "payload":{
+                  "status":"error",
+                  "response":{
+                     "reason":"bad reply"
+                  }
+               }
+            }
+        "#;
+        let actual_reply: Payload<(), ()> = serde_json::from_str(actual_reply).unwrap();
+        let expected_reply = Payload::<(), ()>::Reply(ReplyMessage::PhxReply(PhxReply::Error(
+            ErrorInfo::Reason(Reason::Unknown(UnknownError("bad reply".to_string()))),
+        )));
+        assert_eq!(actual_reply, expected_reply);
+    }
+
+    #[test]
+    fn create_log_sink_error_response() {
+        let json = r#"{"event":"phx_reply","ref":1,"topic":"client","payload":{"status":"error","response":"disabled"}}"#;
```

Comment:
@AndrewDryga The format here seems to differ from the other error responses. Is it possible to align that?

I think the same applies to the "offline" error.

Reply:
This is standard error response by Phoenix

Reply:
Does that mean the other format with `reason` is non-standard? Can we use the standard format everywhere?